spring-grpc-spring-boot-autoconfigure/src/main/java/org/springframework/grpc/autoconfigure/server/GrpcServerAutoConfiguration.java

```diff
@@ -69,4 +73,15 @@ ServerBuilderCustomizers serverBuilderCustomizers(ObjectProvider<ServerBuilderCu
 		return new ServerBuilderCustomizers(customizers.orderedStream().toList());
 	}
 
+	@ConditionalOnMissingBean
```

Comment:
We do not want to impose micrometer on user's. Only if they decide to pull Micrometer onto their classpath should we react to that. Let's remove this `MeterRegistry` bean.

Also, we will want to split this off into its own auto-configuration (e.g. `GrpcServerMetricAutoConfiguration`) and then guard its loading based on whether user has Micrometer on the classpath. The gRPC ecocystem h as a great example [here](https://github.com/yidongnan/grpc-spring-boot-starter/blob/242ee88dc8bc651479a8a8bb7c03ff0ec65aacbb/grpc-server-spring-boot-autoconfigure/src/main/java/net/devh/boot/grpc/server/autoconfigure/GrpcServerMetricAutoConfiguration.java#L57-L77).